Fix release notes (#59566)
tensorflow/tensorflow: RELEASE.md

*   `tf.function`:

    *   `tf.function` now uses the Python inspect library directly for parsing the signature of the Python function it is decorated on.

>     *   This can break certain cases that were previously ignored where the signature is malforme, such as:

Author: ```suggestion
    *   This can break certain cases that were previously ignored where the signature is malformed, such as:
```

        *   Using `functools.wraps` on a function with different signature
        *   Using `functools.partial` with an invalid `tf.function` input
    *   `tf.function` now enforces input parameter names to be valid Python identifiers. Incompatible names are automatically sanitized similarly to existing SavedModel signature behavior.

>     *   Parameterless tf.functions are assumed to have an empty input_signature instead of an undefined one even if the input_signature is unspecified.

Author: ```suggestion
    *   Parameterless `tf.function`s are assumed to have an empty `input_signature` instead of an undefined one even if the `input_signature` is unspecified.
```

    *   `tf.types.experimental.TraceType` now requires an additional `placeholder_value` method to be defined.
    *   `tf.function` now traces with placeholder values generated by TraceType instead of the value itself.

*   Experimental APIs `tf.config.experimental.enable_mlir_graph_optimization` and `tf.config.experimental.disable_mlir_graph_optimization` were removed.

> *   `tf.keras`

Author: ```suggestion
*   `tf.keras`:
```

> *   `tf.SavedModel`

Author: ```suggestion
*   `tf.SavedModel`:
```
